### PR TITLE
xdp-tools: init at 1.2.5

### PR DESCRIPTION
**Pull request by RealTirex**
###### Description of changes

Add xdp-tools package to repository.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/xdp-tools/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{ lib, stdenv, fetchFromGitHub, libbpf, elfutils, libelf, zlib, libpcap
+, clang, llvm, gnumake, gcc, pkgconfig, m4, emacs-nox, wireshark-cli
+}:
```
Comment (ocfox) [inline pkgs/tools/networking/xdp-tools/default.nix:3]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, libbpf
, elfutils
, libelf
, zlib
, libpcap
, clang
, llvm
, gnumake
, gcc
, pkgconfig
, m4
, emacs-nox
, wireshark-cli
}:
```

File: pkgs/tools/networking/xdp-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+let
```
Comment (ocfox) [inline pkgs/tools/networking/xdp-tools/default.nix:5]: Normally, `let in` is not used. You can write directly in the `stdenv.mkDerivation`.So `projectSha256` is not necessary either.

File: pkgs/tools/networking/xdp-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+stdenv.mkDerivation rec {
+  pname = "xdp-tools";
+  inherit version src;
+
+  nativeBuildInputs = [
+    libbpf
+    elfutils
+    libelf
+    zlib
+    libpcap
```
Comment (ocfox) [inline pkgs/tools/networking/xdp-tools/default.nix:25]: `libbpf` `libelf` `libpcap` and some others should not belong to `nativeBuildInputs`. Put them into `BuildInputs`.

File: pkgs/tools/networking/xdp-tools/default.nix [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  FORCE_SYSTEM_LIBBPF = 1;
+  FORCE_EMACS = 1;
+
+  configurePhase = ''
+    ./configure
+  '';
+
+  buildPhase = ''
+    make V=1
+  '';
```
Comment (ocfox) [inline pkgs/tools/networking/xdp-tools/default.nix:48]: Is this necessary?

**Review by ocfox — CHANGES_REQUESTED** [review]
And don't need four commits, maintainers add and init are enough.

File: pkgs/tools/networking/xdp-tools/default.nix (hunk 2 above)
Comment (NoneTirex) [reply pkgs/tools/networking/xdp-tools/default.nix:5]: I added `projectSha256` and `version` to expose these values, easily for the future changes. I understand that `version` variable is used correctly, but I should move `src` directly to the `stdenv.mkDerivation`?

File: pkgs/tools/networking/xdp-tools/default.nix (hunk 4 above)
Comment (NoneTirex) [reply pkgs/tools/networking/xdp-tools/default.nix:48]: You're right, also `configurePhase` is not necessary

File: pkgs/tools/networking/xdp-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  DYNAMIC_LIBXDP = 1;
+  FORCE_SYSTEM_LIBBPF = 1;
+  FORCE_EMACS = 1;
+
+  installPhase = ''
+    export PREFIX=$out
+
+    make install
+  '';
+
```
Comment (ocfox) [inline pkgs/tools/networking/xdp-tools/default.nix:59]:
```suggestion
```

File: pkgs/tools/networking/xdp-tools/default.nix [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    m4
+    emacs-nox
+    wireshark-cli
+  ];
+
+  BPF_CFLAGS = "-fno-stack-protector -Wno-error=unused-command-line-argument";
+  PRODUCTION = 1;
+  DYNAMIC_LIBXDP = 1;
+  FORCE_SYSTEM_LIBBPF = 1;
+  FORCE_EMACS = 1;
```
Comment (ocfox) [inline pkgs/tools/networking/xdp-tools/default.nix:52]: It would be nice if you could add a comment to this line.

**Review by ocfox — CHANGES_REQUESTED** [review]

**Comment by ocfox** [comment]
Does `wireshark-cli` and `gcc`  really needed, I can also build without them.

**Comment by NoneTirex** [comment]
@ocfox i didn't tried without them before, because I was following on xdp-tools spec for package rpm: https://github.com/xdp-project/xdp-tools/blob/master/packaging/rpm/xdp-tools.spec#L11
I was thinking about checking what is needed and what is not, but as a last resort I forgot to check.

File: pkgs/tools/networking/xdp-tools/default.nix [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+  installPhase = ''
+    export PREFIX=$out
+
+    make install
+  '';
```
Comment (ocfox) [inline pkgs/tools/networking/xdp-tools/default.nix:58]:
```suggestion
  makeFlags = [ "PREFIX=$(out)" ];
```
This can be simpler.

**Comment by vcunat** [comment]
So, I dared to "suggest" changes by pushing into your branch.

I'll be needing `libxdp` for `knot-dns` after `libbpf` gets updated in PR #187978, which is why I looked at the expression carefully and added myself as (another) maintainer.  (I had some local WIP version even before you submitted this PR.)

File: pkgs/tools/networking/xdp-tools/default.nix (hunk 7 above)
Comment (vcunat) [reply pkgs/tools/networking/xdp-tools/default.nix]: Done.

File: pkgs/tools/networking/xdp-tools/default.nix (hunk 6 above)
Comment (vcunat) [reply pkgs/tools/networking/xdp-tools/default.nix]: Done.

**Review by vcunat — CHANGES_REQUESTED** [review]
The maintainer list says at the top that it should be sorted alphabetically, so please amend that commit.

File: pkgs/tools/networking/xdp-tools/default.nix (hunk 1 above)
Comment (vcunat) [reply pkgs/tools/networking/xdp-tools/default.nix]: Done.

**Comment by NoneTirex** [comment]
@vcunat your changes are good, and thanks for your suggestions.

**Pull request merged**